packages/logo-grid/index.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useRef, useState } from 'react'
+import Image from '@hashicorp/react-image'
+import Button from '@hashicorp/react-button'
+import fragment from './fragment.graphql'
+import classNames from 'classnames'
+import DialogTooltip from './partials/dialog-tooltip'
+import { useRect } from '@reach/rect'
+import s from './style.module.css'
+
+interface CompanyLogo {
+  /** url of the image */
+  url: string
+  /** format of the image, like "jpg" or "svg" */
+  format: string
+  /** alt text for the image. defaults to company name. */
+  alt?: string
+}
+
+interface Company {
+  /** Company name */
+  name: string
+  /** Short description of what the company offers */
+  description: string
+  /** Optional integration page to link to. */
+  integrationPage?: {
+    /** Slug to link to, to be constructed into a `.com/integrations/{slug}` url. */
+    slug: string
+  }
+  /** Company website address. */
+  link?: string
+  /** Color logo image. */
+  logo?: CompanyLogo
+  /** Monochrome black logo image. */
+  monochromeLogo?: CompanyLogo
+  /** Monochrome white logo image. */
+  whiteLogo?: CompanyLogo
+}
+
+interface LogoGridProps {
+  /** Array of company objects to render as grid items. */
+  data: Array<Company>
+  /** Color scheme for company logos. Ensure all logos have the specified color available. "color" and "monochrome" should be used on light backgrounds. "white" should be used on dark backgrounds. */
+  color?: 'color' | 'white' | 'monochrome'
+  /** If true, when a logo with a company.description is clicked, a tooltip-style dialog  that links to the company website will be shown. */
+  details?: boolean
+  /** If true, links to integration pages are enabled. If an item has a valid `data.integrationPage` property, the logo grid item will be linked. If grid items have both a link and a tooltip, only the link will be used. */
+  integrationLink?: boolean
+  /** If true, borders around logo items will be removed. */
+  removeBorders?: boolean
+  /** Display size of the logos within the grid. */
+  size?: 'small' | 'medium' | 'large'
+  /** Optional className to render on the root element */
+  className?: string
+}
+
+function LogoGrid({
+  data,
+  color = 'color',
+  details,
+  integrationLink,
+  removeBorders,
+  size = 'small',
+  className,
```

Comment:
I've added a `className` prop. Rationale here is  that [a quick SourceGraph search](https://sourcegraph.hashi-mktg.com/search?q=g-logo-grid&patternType=literal) can easily determine whether the old global className was targeted in "overrides" - if so, I immediately add a `className` prop so that we can, for now, retain these overrides rather than having their replacement block rollout of these components.

In cases where there are no overrides, ie that SourceGraph search comes up empty, I could still see a case for adding a `className` prop to enable "safe overrides" (as detailed in the Notion doc on CSS). So if I ever find a component that has 0 overrides in the wild, may still be worth adding this 👍